### ISSUE TITLE
Regression in NC20: no usable file search any more

### ISSUE THREAD

**Issue by andreas-p** [nextcloud/server#24029]
Within hours after upgrading to NC20, I got this user feedback:

> Searching for files is a lot slower than before the update. The result list is very small, and not als clearly laid out as before. Do we need to live with this or is there another solution?

Universal search has several issues:

- Performance  
  This is covered in issue #23835.
- Search context  
  I've been inquiring some very different long-time computer users, and none of them ever used unified search on Windows or MacOS. In practice, there's always a well-known search context, files, contacts or so, neither of them or me could recall a case of "gimme anything with these letters in it". In addition, the usual file search will be like `ls -l <pattern>` or `find . -name <pattern>`, but never `find / -name <pattern>`.
    
- Search result  
  When I'm going to the bakery and ask for "pie", I expect to get offers of apple pie and cherry pie, but not books about baking pies, apps for pies, fotos of the Pietà, all videos with Dizzy Gillespie, and certainly not every time I enter the shop.  
- Result representation  
 The small window is optically very fancy, but in practice just a peekhole on the data. 
 
Summing up, Universal Search is technologically impressive and good for marketing purposes, but not suitable to fulfill every day's need for file search.

Please consider reviving the "old" file search, with options to search the current directory only, and below only. It should work sufficiently on many files, at least in the current directory. The current approach isn't viable.

**Comment by andreas-p** [comment]
@Derkades This is how the file search used to work (and after a delay, it would start a deeper search), which worked fine for the data I'm working with. Right now, I had to disable the file search provider in the code, to prevent unified search from blocking the database server. No way to locate a specific file, in directories with >5k files...

**Comment by PVince81** [comment]
closing in favor of https://github.com/nextcloud/server/issues/23432 which addresses the same issue
